Source/HTTP/Curl/CurlMulti.cpp

```diff
@@ -0,0 +1,157 @@
+#include "pch.h"
+#include "CurlMulti.h"
+#include "CurlProvider.h"
+
+namespace xbox
+{
+namespace http_client
+{
+
+// XCurl doesn't support curl_multi_timeout, so use a small, fixed delay between calls to curl_multi_perform
```

Comment:
should we be supporting this?

Reply:
Yes and no.
It would be great to have an equivalent but `curl_multi_timeout` is very much designed to work with Unix style `select` calls which have the completely unacceptable (in my opinion) downside of *blocking* the thread until an event occurs.

I will oppose any HTTP implementation that causes a thread calling XTaskQueueDispatch to block until the HTTP request is completed

Reply:
It would be nice to have some mechanism to know when there is work to do in curl_multi_perform.  Unless I'm misunderstanding something, the only way the only way to implement parallel http requests in a non-blocking fashion is to add an arbitrary sleep.

Reply:
libcurl unfortunately doesn't provide any way to notify you.  You can poll with curl_multi_poll or curl_multi_wait with a zeroed-out or small timeout, but libcurl is designed with select (or those two functions on windows where select doesn't work on arbitrary file handles) in mind.  That's better than an arbitrary sleep, but it still burns a thread.  You don't realize the joys of overlapped IO until its missing :/

Reply:
Indeed, the solution here at least doesn't burn the thread since we're scheduling work in the future.

I would still love for XCurl to provide a GDK specific function that integrates curl polling into the XAsync framework nicely

Reply:
Going to leave the sleep for now - really a priority to avoid burning the thread.

Reply:
Just to be clear, there is no sleep in this code. `XTaskQueueSubmitDelayedCallback` submits the task to be executed in the future but does not send the thread to sleep.